baselinemarkers/get_marker_genes_rnaseq.R

```diff
@@ -0,0 +1,363 @@
+    cols <- which(col_names == name)
+    rowMeans(dt_numeric[, ..cols], na.rm = TRUE)
+})
+
+# Combine into a new data.table with Gene ID
+result_dt <- data.table(`Gene ID` = gene_ids, setNames(as.data.table(averaged_list), unique_names))
+
+# Assuming 'Gene ID' is the first column and the rest are numeric
+# remove Genes with no expression in any column (won't be used for marker identification) 
+result_dt <- result_dt[rowSums(result_dt[, -1, with = FALSE] != 0) > 0]
+dim(result_dt)
+
+# Remove 'Gene ID' column and convert remaining data to matrix
+mat <- as.matrix(result_dt[, !"Gene ID", with = FALSE])
+
+# Set rownames from 'Gene ID'
+rownames(mat) <- result_dt[["Gene ID"]]
+
+####################################
+# run MGFR to get marker genes
+####################################
+
+# Function to detect marker genes using normalized RNA-seq data
+markers.list <- getMarkerGenes.rnaseq(mat, 
+                                      class.vec = colnames(mat),
+                                      samples2compare="all", 
+                                      annotate=FALSE, 
+                                      gene.ids.type="ensembl", 
+                                      score.cutoff=1)
+
+print( names(markers.list) )
+
+############################################################################
+# Sanity check -  that marker genes are not repeated in any list
+############################################################################
+
+# Extract gene IDs from each list and clean them
+gene_lists <- lapply(markers.list, function(x) sub(" :.*", "", x))
+
+# combine all gene IDs into a single vector with list (tissue) names
+gene_with_source <- unlist(gene_lists, use.names = FALSE)
+sources <- rep(names(gene_lists), lengths(gene_lists))
+
+# create a data frame of gene ID and source tissue
+df <- data.frame(gene = gene_with_source, source = sources, stringsAsFactors = FALSE)
+
+# count in how many distinct groups each gene appears
+
+duplicated_genes <- df %>%
+    distinct(gene, source) %>%        # remove within-list duplicates
+    group_by(gene) %>%
+    summarise(n_sources = n()) %>%
+    filter(n_sources > 1)
+
+# Show genes present in more than one list
+print(duplicated_genes)
+
+# Stop execution if duplicates are found
+if (nrow(duplicated_genes) > 0) stop("Error: Duplicated genes found across gene marker lists.")
+
+############################################################################
+# Filter all marker lists by SPECIFICITY_SCORE_CUTOFF
+############################################################################
+markers.list <- lapply(markers.list, function(marker_vec) {
+    # Keep only entries that contain ":"
+    marker_vec <- marker_vec[grepl(":", marker_vec)]
+    
+    # Extract scores
+    scores <- as.numeric(sub(".*: ", "", marker_vec))
+    
+    # Filter out entries with score >= cutoff
+    marker_vec[!is.na(scores) & scores <= SPECIFICITY_SCORE_CUTOFF]
+})
+
```

Comment:
Adding a check after filtering marker genes

```suggestion
total_markers <- sum(sapply(markers.list, length))
if (total_markers == 0) {
  stop("No markers found after applying specificity score cutoff. Consider lowering the cutoff value.", call. = FALSE)
}
```